### PR TITLE
add fund sufficient check in send

### PR DESCRIPTION
**Pull request by deaswang**
add fund sufficient check in send to avoid SysErrInsufficientFunds send. like https://filfox.info/zh/address/f1o2bwv33mqejxeqmh7qtb7wob4pjdho6rgvjp5la from 2020-12-23 14:40:30.

### CODE REVIEW

File: cli/send.go [hunk 1]
```diff
@@ -143,6 +148,23 @@ var sendCmd = &cli.Command{
 			Params:     params,
 		}
 
+		// Funds insufficient check
+		fromBalance, err := api.WalletBalance(ctx, msg.From)
+		if err != nil {
+			return err
+		}
+		totalCost := types.BigMul(msg.GasFeeCap, types.NewInt(uint64(msg.GasLimit)))
+		if msg.Method == builtin.MethodSend {
```
Comment (Kubuxu) [inline cli/send.go:157]: the `msg.Value` is always sent over, no matter if the method is Send or note.

**Review by magik6k — COMMENTED** [review]
Just 1 nitpick

File: cli/send.go [hunk 2]
```diff
@@ -143,6 +148,21 @@ var sendCmd = &cli.Command{
...
 
+		// Funds insufficient check
+		fromBalance, err := api.WalletBalance(ctx, msg.From)
+		if err != nil {
+			return err
+		}
+		totalCost := types.BigAdd(types.BigMul(msg.GasFeeCap, types.NewInt(uint64(msg.GasLimit))), msg.Value)
+
+		if fromBalance.LessThan(totalCost) {
+			fmt.Printf("From balance %s attoFIL less than total cost %s attoFIL\n", fromBalance, totalCost)
```
Comment (magik6k) [inline cli/send.go:159]: I'd do
```suggestion
			fmt.Printf("From balance %s less than total cost %s\n", types.FIL(fromBalance), types.FIL(totalCost))
```

File: cli/send.go [hunk 3]
```diff
@@ -143,6 +148,21 @@ var sendCmd = &cli.Command{
 			Params:     params,
 		}
 
+		// Funds insufficient check
+		fromBalance, err := api.WalletBalance(ctx, msg.From)
+		if err != nil {
+			return err
```
Comment (magik6k) [inline cli/send.go:154]: Maybe ignore this error when force is passed

**Review by magik6k — APPROVED** [review]

**Review by Kubuxu — APPROVED** [review]

File: cli/send.go (hunk 3 above)
Comment (deaswang) [reply cli/send.go]: if force specified will not check insufficient fund

**Review by magik6k — APPROVED** [review]

**Pull request merged**